core/src/main/java/org/apache/druid/data/input/impl/MapInputRowParser.java

```diff
@@ -93,23 +94,32 @@ public static InputRow parse(
     final DateTime timestamp;
     try {
       timestamp = timestampSpec.extractTimestamp(theMap);
-      if (timestamp == null) {
-        final String input = theMap.toString();
-        throw new NullPointerException(
-            StringUtils.format(
-                "Null timestamp in input: %s",
-                input.length() < 100 ? input : input.substring(0, 100) + "..."
-            )
-        );
-      }
     }
     catch (Exception e) {
-      throw new ParseException(e, "Unparseable timestamp found! Event: %s", theMap);
+      throw new ParseException(e, "Unparseable timestamp found! Event: %s", rawMapToPrint(theMap));
+    }
+    if (timestamp == null) {
+      throw new ParseException("Unparseable timestamp found! Event: %s", rawMapToPrint(theMap));
```

Comment:
Instead of printing the whole map, I think it would be better to just print the timestamp string that was unparsable.
Similar comment on line 99, this would get rid of the need of `rawMapToPrint` and guarantees that the invalid timestamp is always logged regardless of where in the event it is.

```suggestion
      throw new ParseException("Unparseable timestamp found! Timestamp column (%s): %s", timestampSpec.getTimestampColumn(), theMap.get(timestampSpec.getTimestampColumn()));
```

Reply:
It is a good idea to print the timestamp string, but I would like to keep the current behavior as well (this logging is not what I added in this PR). I modified the error message to include timestamp.